src/main/sensors/gyro.c

```diff
@@ -161,13 +161,13 @@ void gyroUpdate(void)
     if (gyroLpfCutFreq) {
         if (!gyroFilterStateIsSet) initGyroFilterCoefficients(); /* initialise filter coefficients */
 
-        if (gyroFilterStateIsSet) {
-            for (axis = 0; axis < XYZ_AXIS_COUNT; axis++){
+        for (axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
+            if (gyroFilterStateIsSet) {
                 gyroADCf[axis] = applyBiQuadFilter((float) gyroADC[axis], &gyroFilterState[axis]);
                 gyroADC[axis] = lrintf(gyroADCf[axis]);
+            } else {
+                gyroADCf[axis] = gyroADC[axis]; // Otherwise float pid controller will not have gyro input when filter disabled
             }
-        } else {
-            gyroADCf[axis] = gyroADC[axis]; // Otherwise float pid controller will not have gyro input when filter disabled
```

Comment:
yeah true. Well never mind. We will fix it for 3.0. Anyway nobody ever discovered this as we always have filter on.
I did some testing today and found that